src/data/reddit_dataset.py

```diff
@@ -65,22 +67,22 @@ def __iter__(self) -> Iterator[tuple[str, str]]:
             with open(file, "rt") as f_in:
                 for line in f_in:
                     sample = json.loads(line)
-                    utterances = [self.bos_token + it + self.eos_token for it in sample["thread"]]
+                    utterances = [self.bos_token + it for it in sample["thread"]]
```

Comment:
That is not good for T5 model. This model expect only one bos and eos tokens per sequence. And in this case, you would add bos tokens to each utterance.

Let's update code in this way:
1. Add functions to prepare context and target based on tokenizer name (`prepare_blenderbot`, `prepare_t5`)
2. In this function only work with raw utterances and select for current sample
3. Before yielding them, pass to functions from first step